contribs/gmf/src/directives/search.js

```diff
@@ -0,0 +1,281 @@
+   * TODO must be definable by an integrator
+   * @type {Array<string>}
+   * @private
+   */
+  this.datasources_ = [{
+    dataTitle: 'From demo 1.6',
+    url: 'http://geomapfish-demo.camptocamp.net/1.6/wsgi/fulltextsearch'
+  }, {
+    dataTitle: 'From Cartoriviera - ',
+    url: 'http://map.cartoriviera.ch/main/wsgi/fulltextsearch',
+    groupByProperties: ['04rue', '03adresse']
+  }];
+
+  /**
+   * @type {TypeaheadOptions}
+   * @export
+   */
+  this.options = /** @type {TypeaheadOptions} */ ({
+    highlight: true,
+    hint: undefined,
+    minLength: undefined
+  });
+
+  /**
+   * @type {Array.<TypeaheadDataset>}
+   * @export
+   */
+  this.datasets = [];
+  for (var i = 0; i < this.datasources_.length; i++) {
+    var datasource = this.datasources_[i];
+    var groupByProperties = datasource['groupByProperties'] || [];
+    var filter;
+    do {
+      //Add an optional filter function to keep objects only from one
+      //"layername" from a GMF's fulltextsearch service.
+      var dataTitle = datasource['dataTitle'];
+      if (groupByProperties[0]) {
+        filter = this.filterLayername_(groupByProperties[0]);
+        dataTitle = dataTitle + groupByProperties[0];
+        groupByProperties.shift();
+      } else {
+        filter = null;
+      }
+
+      this.datasets.push(this.createDataset_(datasource['url'],
+                                             filter,
+                                             dataTitle));
+    }
+    while (groupByProperties.length > 0);
+  }
+
+
+
+  /**
+   * @type {ngeox.SearchDirectiveListeners}
+   * @export
+   */
+  this.listeners = /** @type {ngeox.SearchDirectiveListeners} */ ({
+    selected: angular.bind(this, gmf.SearchController.selected_)
+  });
+
+};
+
+
+/**
+ * @return {ol.layer.Vector} The vector layer.
+ * @private
+ */
+gmf.SearchController.prototype.createVectorLayer_ = function() {
+  var vectorLayer = new ol.layer.Vector({
+    source: new ol.source.Vector()
+  });
+  // Use vectorLayer.setMap(map) rather than map.addLayer(vectorLayer). This
+  // makes the vector layer "unmanaged", meaning that it is always on top.
+  vectorLayer.setMap(this.map);
+  return vectorLayer;
+};
+
+
+/**
+ * @param {string} url A ws url that return geojson
+ * @param {Function?} filter A filter based on a featuresCollection's array
+ * @param {string} dataTitle the tile of this dataset's results
+ * @return {TypeaheadDataset} a typeahead dataset.
+ * @private
+ */
+gmf.SearchController.prototype.createDataset_ = function(url, filter,
+                                                             dataTitle) {
+  var rootScope = this.rootScope_;
+  var compile = this.compile_;
+  var bloodhoundEngine = this.createAndInitBloodhound_(url, filter);
+  return /** @type {TypeaheadDataset} */ ({
+    source: bloodhoundEngine.ttAdapter(),
+    display: function(suggestion) {
+      var feature = /** @type {ol.Feature} */ (suggestion);
+      return feature.get('label');
+    },
+    templates: {
```

Comment:
You need to cast the object here, to make sure Closure recognizes it as a `TypeaheadTemplates`.